src/Model/Comment.php

```diff
@@ -880,6 +889,30 @@ public function ReplyForm()
         return $controller->ReplyForm($this);
     }
 
+    public function UpdateForm()
+    {
+        if (!$this->canUpdate()) {
+            return null;
+        }
+        // Check parent is available
+        $parent = $this->Parent();
+
+        if (!$parent || !$parent->exists()) {
+            return null;
+        }
+
+        // Build update controller
+        $extended_controller = $this->extend('updateController');
+        $extended_controller = $extended_controller ? $extended_controller[0] : null;
```

Comment:
Can you please explain what is this doing and why we can't create the `CommentingController` straight away like in the `ReplyForm` method above?

Reply:
hello michal, thank you for your review

I've added comments blocks where missing. For this comment: I am also using silverstripe-reviews which build on top of this extension. This extension was the way I found to use the controller from they, which is this form + new controllers.

Reply:
I think this is the wrong way around this. Even getting the controller just to produce a form to then backfill with data here and pass it back.. I guess it's how it was done for the standard 'post comment' form.

There are already several hooks to update the form itself (e.g. `updateUpdateForm` or `alterCommentForm` within the controller class, so I don't think another one needs to be here.